Symbiota/Symbiota
# Wkt to geojson

# Issues #836 #307 

### Note This is a 3.2 issue and will not be merge until we are ready to make Development 3.2 staging.

# Problem Summary
Our WKT, Well Known Text, format is incorrectly ordered in latitude then longitude. This means that wkt's created from would not work if imported into our system. Also if some of the geometries are getting transferred to lng lat then all points and spacial indices need to be changed to lng lat in order to use Mysql/MariaDB spacial functions.

# Work Summary
- Fully deprecate use of `footprintWkt` attribute from `fmchecklists` table and swap to using footprintGeoJson
- Swap all spacial data to be lng lat. (Only omocurrpoints, geothesaurus was build for geoJson)
- Integrate all map systems were drawing wkts to draw GeoJson.
 - Map 
- Remove WKT from variable, function, input names and ids (still on going see `More Work` section)

# More Work
I am starting the review process early since this is a large pr and will likely have issues but there is still some more work I know off.
- Finishing removing straggling references to WKT in function names, variables, ids, input names
- Remove unused WKT utilities

# Running Locally
There is no extra symbini config you just need to run the 3.2 patch sql. This only adds things minus some data cleaning of omoccurpoints that have no lat or lng data which should not interfere with any 3.1 code.

# How to Test
- Map Search
  - Check Than all shape searches work
- Regular Search
  - Rectangle Search Works
  - Circle Search Works
  - Polygon search works. Output in the search box is geojson
- Checklist Polygon Tool Works as expected
- Any other polygon tool I have missed should work and use geojson

# Pull Request Checklist:

# Pre-Approval

- [x] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [x] Hotfixes should be branched off of the `master` branch and **squash and merged** back into the `master` branch.
- [x] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [x] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
- [x] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [x] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [x] Comment which GitHub issue(s), if any does this PR address
- [x] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).

# Post-Approval

- [ ] It is the code author's responsibility to merge their own pull request after it has been approved
- [ ] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [ ] If this PR represents a hotfix into the `master` branch, a subsequent PR from `master` into `Development` should be made **merge** option (i.e., no squash).
- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [ ] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!
